src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TagHelperResolverFactory.Resolver.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.AspNetCore.Razor.Serialization;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+
+namespace Microsoft.CodeAnalysis.Razor;
+
+internal partial class TagHelperResolverFactory
+{
+    private sealed class Resolver(ITelemetryReporter telemetryReporter) : ITagHelperResolver
+    {
+        private readonly CompilationTagHelperResolver _resolver = new(telemetryReporter);
+
+        public ValueTask<TagHelperResolutionResult> GetTagHelpersAsync(
+            Project workspaceProject,
+            IProjectSnapshot projectSnapshot,
+            CancellationToken cancellationToken = default)
+        {
+            if (workspaceProject is null)
```

Comment:
nit: since these are only called by our code, and we have nullable enabled, can we remove these checks?

Reply:
That's likely. I'll take a look.